src/test/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandlerTest.java

```diff
@@ -2157,37 +2092,140 @@ public void getQueueSpeedMatchingPartlyFilter(TestContext context) throws Except
             async.countDown();
         });
         async.awaitSuccess();
-        context.assertTrue(waitForQueueSpeed("speed_a.*",1));
+        assertQueueSpeed(context, "speed_a.*", 1);
         consumer.unregister();
     }
 
+    /**
+     * Helper method to wait for a particular queue state. Parameters provided with null (except
+     * context) will not be taken into account for the queue state comparison
+     *
+     * @param context               The Test context to be used
+     * @param queuesSize            The expected size of the returned statistics queues
+     * @param element               The element in the retrieved statistics queue array to be
+     *                              checked against
+     * @param queueName             The name of the particular expected queue at position=element
+     * @param queueSize             The size of the particular expected queue at position=element
+     * @param queueFailures         The failures of the particular expected queue at
+     *                              position=element
+     * @param queueSlowdownTime     The slowDownTime of the particular expected queue at
+     *                              position=element
+     * @param queueBackpressureTime The backpressureTime of the particular expected queue at
+     *                              position=element
+     * @oaram filter                The filter to be applied to the statistic queues request
+     */
+    private void assertQueueState(TestContext context, String filter, Integer queuesSize,
```

Comment:
Minor, maybe annotate the params that can be null with @Nullable

Reply:
Will do.